parser/src/toMachineConfig.ts

```diff
@@ -0,0 +1,230 @@
+import {
+  Actions,
+  MachineConfig,
+  StateNodeConfig,
+  TransitionConfigOrTarget,
+} from "xstate";
+import { MaybeArrayOfActions } from "./actions";
+import { TMachineCallExpression } from "./machineCallExpression";
+import { StateNodeReturn } from "./stateNode";
+import { MaybeTransitionArray } from "./transitions";
+import { GetParserResult } from "./utils";
+
+export interface ToMachineConfigParseOptions {
+  /**
+   * Whether or not to hash inline implementations, which
+   * allow for parsing inline implementations as code.
```

Comment:
q: when do we not want this?

Reply:
We don't hash inline implementations for the viz, since we don't support them (and they're not necessary) so they'd show a bunch of horrible `123jha` ids in the actions/services.

Reply:
What does it mean for the viz though? Can machines with inline implementation be visualized? Couldn't we just display those as "anonymous" in the viz?

Reply:
I remember why I didn't do this here - again, it's to do with the existing test setup in @xstate/machine-extractor. Let's address this as a followup PR?